features/step_definitions/search_steps.rb

```diff
@@ -29,3 +29,41 @@
     page.should have_css('dt a span.match', text: text)
   end
 end
+
+Then /^they can filter the table by (?:name|system name)$/ do
+  @items = find_items
```

Comment:
An instance variable suggests it's going to be used and/or changed elsewhere. However, this variable belongs to this method only.
```suggestion
  all_items = find_items
```